From af8fc2e958f6d43c2db7efb774c77e8743703eb2 Mon Sep 17 00:00:00 2001
From: Jiaxing Shi <quic_jiaxshi@quicinc.com>
Date: Wed, 15 Mar 2023 16:20:01 +0800
Subject: [PATCH] Remove try_run for cross compiling

try_run: When cross compiling, the executable compiled in the first
step usually cannot be run on the build host.

Signed-off-by: Jiaxing Shi <quic_jiaxshi@quicinc.com>
---
 src/cpp/CMakeLists.txt | 34 ----------------------------------
 1 file changed, 34 deletions(-)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index a5213c25..7346aa93 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -388,41 +388,7 @@ if(NOT CMAKE_ATOMIC_LIB)
     set(CMAKE_ATOMIC_LIB)
 endif()
 
-try_run(SM_RUN_RESULT SM_COMPILE_RESULT
-        "${CMAKE_CURRENT_BINARY_DIR}/shmtest"
-        "${CMAKE_SOURCE_DIR}/cmake/modules/check_shared_mutex_priority.cpp"
-        LINK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_ATOMIC_LIB}
-        RUN_OUTPUT_VARIABLE SM_RUN_OUTPUT)
-
-if(SM_COMPILE_RESULT AND NOT SM_RUN_RESULT)
-    string(STRIP ${SM_RUN_OUTPUT} SM_RUN_OUTPUT)
-    message(STATUS "Framework's shared_mutex is ${SM_RUN_OUTPUT}")
-endif()
-
-if(SM_RUN_OUTPUT STREQUAL "PTHREAD_RWLOCK_PREFER_READER_NP")
-    set(USER_CAN_CHOOSE_SHARED_MEMORY_THIRDPARTY ON)
-else()
-    message(STATUS "Forcing third party shared_mutex")
-    set(USE_THIRDPARTY_SHARED_MUTEX ON)
-endif()
 
-cmake_dependent_option(
-    USE_THIRDPARTY_SHARED_MUTEX [=[
-Forces the use of a Boost-based shared_mutex implementation
-instead of the framework one. Useful to cope with issues on
-framework implementations like misguided sanitizer reports.
-This implementation will be used by default on frameworks
-lacking the shared_mutex feature like those not fulfilling
-C++17.
-]=] OFF
-    "USER_CAN_CHOOSE_SHARED_MEMORY_THIRDPARTY"
-    ON)
-
-unset(USER_CAN_CHOOSE_SHARED_MEMORY_THIRDPARTY)
-unset(SM_RUN_RESULT)
-unset(SM_COMPILE_RESULT)
-unset(SM_RUN_OUTPUT)
-unset(CMAKE_ATOMIC_LIB)
 
 # Generate the proper configure file
 configure_file(${PROJECT_SOURCE_DIR}/include/${PROJECT_NAME}/config.h.in
-- 
2.25.1

